clients/itwin/src/Client.ts

```diff
@@ -83,6 +89,19 @@ export abstract class Client {
       return this._url;
     }
 
+    if (this.baseUrl) {
+      const prefix = Config.App.query("imjs_url_prefix");

```

Comment:
Can you elaborate how this is a breaking change, and what do you call "users that are relying on Buddi"? How are they relying on it?

As mentioned previously, I made the change for imodeljs-config repo as well, to include `imjs_url_prefix` value that is consistent with other values in the configuration.

The whole point of this was to switch all imodeljs traffic to use APIM endpoints, and now you're telling to use APIM endpoints only sometimes. My previous implementation (before `imjs_url_prefix`) allowed setting URLs for any client, for any environment, and would have also allowed tweaking the end-application configuration for the particular needs (and change URLs back to non-APIM, though again, I don't see a point in that).

Reply:
The current client libraries are configured to get their URL using BUDDI and by default they will get the one available for the region `0`.  If the user of the package wants to switch to a different environment, they need to use the `imjs_buddi_resolve_url_using_region` for the appropriate region code. Which is currently something every Bentley application will be doing.

With this change, that workflow will no longer work for all clients since it does not take into account `imjs_buddi_resolve_url_using_region` if `baseUrl` is defined. Therefore changing the behavior for consumers of the client libraries.

The imodeljs-config repo change doesn't help any of the consumers of the packages/clients, it only helps the devs on the iModel.js team.

Instead of falling back to the old APIs, we could add the handling of `imjs_buddi_resolve_url_using_region` for the main 2 variables, `102` and `103`. It's not ideal but would avoid breaking anyone that is currently relying on it and allow most traffic to go through APIM.

Reply:
@laurynasr I'll approve this as-is for now and we'll potentially follow up in a separate PR.  I want to get this to start  being tested before the next release 2.14.